### ISSUE TITLE
Strange JSON.stringify() behaviour

### ISSUE THREAD

**Issue by cesco69** [oven-sh/bun#15056]
### What version of Bun is running?

1.1.34+5e5e7c60f

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

Execute JSON.stringify() on an object and see the resulting string.

The object itself doesn't matter... but let's use the following object for example (the same I used while saw this bug):
```javascript
const MY_OBJ = { a: "b" }
```

### What is the expected behavior?

As Node.js do, the expected result should be:

```javascript
JSON.stringify(MY_OBJ);
// Output ==> '{"a":"b"}'
```

### What do you see instead?

Executing the same code into Bun:

```javascript
JSON.stringify(MY_OBJ);
// Output ==> "{\"a\":\"b\"}"
```

### Additional information

TLDR:
- Node.js doesn't escapes double quotes
- Bun escapes automatically double quotes

I think that they should have the same behaviour, in order to don't occur in errors.

**Comment by ippsav** [comment]
This bug seems to be windows only
